src/ast_render.cpp

```diff
@@ -352,7 +352,7 @@ static void string_literal_escape(Buf *source, Buf *dest) {
         } else if (is_printable(c)) {
             buf_append_char(dest, c);
         } else {
-            buf_appendf(dest, "\\x%x", (int)c);
+            buf_appendf(dest, "\\x%02x", (int)c);
```

Comment:
Is this correct?

>   Precision
>       An optional precision, in the form of a period ('.')   followed  by  an
>       optional  decimal  digit string.

Reply:
`02` doesn't specify precision, it's padding:
```c
$ cat test.c 
#include <stdio.h>
int main()
{
	printf("\\x%x\n", 12);
	printf("\\x%02x\n", 12);
	return 0;
}
$ cc -o test test.c 
$ ./test 
\xc
\x0c
```